### PR TITLE
testqgs3drendering: Add a test for shadows effect

### PR DESCRIPTION
**Pull request by ptitjano**

### CODE REVIEW

File: tests/src/3d/testqgs3drendering.cpp [hunk 1]
```diff
@@ -1525,6 +1526,55 @@ void TestQgs3DRendering::testFilteredExtrudedPolygons()
...
+  // When running the test on Travis, it would initially return empty rendered image.
+  // Capturing the initial image and throwing it away fixes that. Hopefully we will
+  // find a better fix in the future.
+  Qgs3DUtils::captureSceneImage( engine, scene );
+  QImage img = Qgs3DUtils::captureSceneImage( engine, scene );
+
+  delete mapSettings;
+  delete scene;
+
+  QGSVERIFYIMAGECHECK( "shadow_effect", "shadow_effect", img, QString(), 40, QSize( 0, 0 ), 2 );
```
Comment (nyalldawson) [inline tests/src/3d/testqgs3drendering.cpp:1575]: I'd suggest using a solid color terrain here if possible -- it'd make it easier to see exactly what the test is checking

Comment (ptitjano) [reply tests/src/3d/testqgs3drendering.cpp:1575]: I only opened this PR to check if the shadows effect works in qt6. I can't make it work locally and it doesn't work on the ci too. The shadows are not visible.

**Comment by ptitjano** [comment]
@nyalldawson https://github.com/qgis/QGIS/issues/58184

**Comment by nyalldawson** [comment]
@ptitjano what's the status here? is this still a draft? if so, can we close till it's ready please?

**Comment by ptitjano** [comment]
> @ptitjano what's the status here? is this still a draft? if so, can we close till it's ready please?

The shadows effect does not work in qt6. I tried to understand why but It looks like a very complicated issue. So, yes, let's close it at the moment.

**Pull request closed without merging**